xlators/mgmt/glusterd/src/glusterd-syncop.c

```diff
@@ -2560,10 +2624,31 @@ gd_sync_task_begin(dict_t *op_ctx, rpcsvc_request_t *req)
     uint32_t op_errno = 0;
     gf_boolean_t cluster_lock = _gf_false;
     time_t timeout = 0;
+    glusterd_peerinfo_t *peerinfo = NULL;
+    char *af = NULL;
 
     conf = this->private;
     GF_ASSERT(conf);
 
+    ret = dict_get_str(this->options, "transport.address-family", &af);
+    if (ret || af == NULL)
+        af = "inet";
+    RCU_READ_LOCK;
+    cds_list_for_each_entry_rcu(peerinfo, &conf->peers, uuid_list)
+    {
+        if (peerinfo->connected == 0)
+            continue;
+        struct sockaddr sockaddr = {0};
+        if (!strcmp("inet6", af))
+            sockaddr.sa_family = AF_INET6;
+        else
+            sockaddr.sa_family = AF_INET;
+        ret = gd_try_connect(peerinfo->hostname, peerinfo->port, &sockaddr);
```

Comment:
I am not sure if this is a right approach to resolve it. Did you try to debug why the other peer nodes(glusterd) are not getting disconnect event?

Reply:
Hello, @mohit84,
It seems that it is difficult for glusterd to imediately perceive an abnormal disconnection event of nodes, since there is no interaction between the node and glusterd when node drops. In this case, once a node drops, glusterd can only detect whether it is still alive through the probe mechanism,if keepalive operation is turned on. Depending on the configuration, this process may take from a few seconds to a few minutes.